Handle "repo --gpgkey" in kickstart to verify downloaded packages

Especially important for lorax - they will not verified in any way without
setting yum options here.

Signed-off-by: Marek Marczykowski <marmarek@invisiblethingslab.com>
Index: pungi-3.03/src/pypungi/__init__.py
===================================================================
--- pungi-3.03.orig/src/pypungi/__init__.py
+++ pungi-3.03/src/pypungi/__init__.py
@@ -242,7 +242,7 @@
 
     def _add_yum_repo(self, name, url, mirrorlist=False, groups=True,
                       cost=1000, includepkgs=None, excludepkgs=None,
-                      proxy=None):
+                      proxy=None, gpgkey=None):
         """This function adds a repo to the yum object.
         name: Name of the repo
         url: Full url to the repo
@@ -279,6 +279,10 @@
         thisrepo.exclude = excludepkgs
         thisrepo.includepkgs = includepkgs
         thisrepo.cost = cost
+        if gpgkey:
+            thisrepo.gpgcheck = True
+            thisrepo.gpgkey = yum.parser.varReplace(gpgkey,
+                                                    self.ayum.conf.yumvar)
         # Yum doesn't like proxy being None
         if proxy:
             thisrepo.proxy = proxy
@@ -309,6 +313,7 @@
         yumconf.installroot = os.path.join(self.workdir, 'yumroot')
         yumconf.uid = os.geteuid()
         yumconf.cache = 0
+        yumconf.assumeyes = True
         yumconf.failovermethod = 'priority'
         yumconf.deltarpm = 0
         yumvars = yum.config._getEnvVar()
@@ -339,7 +344,8 @@
                                    cost=repo.cost,
                                    includepkgs=repo.includepkgs,
                                    excludepkgs=repo.excludepkgs,
-                                   proxy=repo.proxy)
+                                   proxy=repo.proxy,
+                                   gpgkey=repo.gpgkey)
             else:
                 self._add_yum_repo(repo.name, repo.baseurl,
                                    mirrorlist=False,
@@ -347,7 +353,8 @@
                                    cost=repo.cost,
                                    includepkgs=repo.includepkgs,
                                    excludepkgs=repo.excludepkgs,
-                                   proxy=repo.proxy)
+                                   proxy=repo.proxy,
+                                   gpgkey=repo.gpgkey)
 
         self.logger.info('Getting sacks for arches %s' % self.valid_arches)
         self.ayum._getSacks(archlist=self.valid_arches)
